FogCreek/Glitch-Community
# Obtainable dew: Collection Item

## Links
* Remix link: https://obtainable-dew.glitch.me

## GIF/Screenshots:
![Collection Item on Profile Page](https://user-images.githubusercontent.com/519336/56435329-118c2080-62a6-11e9-97a3-92cdefbd4560.png)
Collection Item on Profile Page

![Collection Item on Project Page](https://user-images.githubusercontent.com/519336/56435344-17820180-62a6-11e9-800e-3fc301872be9.png)
Collection Item on Project Page

## Changes:
Also created a few subcomponents
* Added `privacy` badge:
![image](https://user-images.githubusercontent.com/519336/56436332-6f6e3780-62a9-11e9-823f-f16c59ba84cf.png)

* Created `project-item-small` component
![image](https://user-images.githubusercontent.com/519336/56435439-6a5bb900-62a6-11e9-946c-7eea0d43b790.png)

* Also added back the dropshadow for buttons within items (project, teams, users, and collections):
![dropdown](https://user-images.githubusercontent.com/519336/56435900-1ce04b80-62a8-11e9-99c3-9578f51a6bed.gif)
My rationale: the hover background color for buttons is really subtle, especially for collection items when compared to the hover state for projects within the collection card.  

* Cleaned up storybook a bit too and consolidate Project and Collection Items

## How To Test:
* Check collection cards on these pages:
* Project page: https://obtainable-dew.glitch.me/~bongo-cat
    * should see author avatar displayed
* Your own profile page: https://obtainable-dew.glitch.me/@scientiffic
    * check that dropdown is still accessible for deleting collection if you're authorized
* Someone else's profile page: https://obtainable-dew.glitch.me/@SteGriff
* A team page: https://obtainable-dew.glitch.me/@community
* Empty collections: https://obtainable-dew.glitch.me/@empty-team

## Feedback I'm looking for:
* Edited markdown length to try to prevent collection descriptions from appearing cut off. doesn't work in all cases, but in most - maybe there's another approach?
![image](https://user-images.githubusercontent.com/519336/56436246-2ddd8c80-62a9-11e9-95a1-b6947b60d7ad.png)
* For the private badge, I created it by setting empty children - maybe there's some other way to bypass this required prop?
* Also fixed render of collection descriptions for dark collection colors:
![image](https://user-images.githubusercontent.com/519336/56436911-86158e00-62ab-11e9-94f0-974ece5e4035.png)
